From: Lipeng <lipeng321@huawei.com>
Date: Mon, 23 Oct 2017 19:51:06 +0800
Subject: net: hns3: remove redundant memset when alloc buffer
Patch-mainline: v4.15-rc1
Git-commit: 51145dae2748233315a7a411cd97f4bedf8cc22f
References: bsc#1104353 FATE#326415

HW will use packet length to write packets to buffer or read
packets from buffer. There is a redundant memset when alloc buffer,
the memset have no sense and will increase time-consuming.
This patch removes it.

Fixes: 76ad4f0 (net: hns3: Add support of HNS3 Ethernet Driver for hip08 SoC)

Signed-off-by: Lipeng <lipeng321@huawei.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/hisilicon/hns3/hns3pf/hns3_enet.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/net/ethernet/hisilicon/hns3/hns3pf/hns3_enet.c
+++ b/drivers/net/ethernet/hisilicon/hns3/hns3pf/hns3_enet.c
@@ -1422,8 +1422,6 @@ static int hns3_alloc_buffer(struct hns3
 	cb->length = hnae_page_size(ring);
 	cb->type = DESC_TYPE_PAGE;
 
-	memset(cb->buf, 0, cb->length);
-
 	return 0;
 }
 
